charts/castai-cluster-controller/templates/deployment.yaml

```
@@ -67,9 +68,20 @@ spec:
       nodeSelector:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.affinity }}
       affinity:
-        {{- toYaml . | nindent 8 }}
+        nodeAffinity:
+          {{- with .Values.nodeAffinity }}
```

if we don't change the affinity in values something like that should work 
```suggestion
          {{- with .Values.affinity.nodeAffinty  }}
```

of course it would be good to check if affinity is set first but I just wanted to show the idea